### ISSUE TITLE
Ingredients not being hidden from JEI

### ISSUE THREAD

**Issue by justinrusso** [mezz/JustEnoughItems#1482]
I am trying to use ItemStages to stage enchanting books. When staging enchantment books, the books end up staged but continue to show in JEI even with all stages cleared. This originally worked (SevTech 3.0.8 hides them perfectly fine), but attempting to update to the newer versions of JEI cause this issue.

I'm lead to believe this is caused by a change in JEI. I've enabled debug logging for JEI and in the debug log it shows
```
[20:21:23] [main/DEBUG] [jei]: Removed ingredient: 1xitem.enchantedBook@0 minecraft:enchanted_book nbt:{StoredEnchantments:[{lvl:1s,id:0s}]}
```

I've noticed that compared to 4.13.1.223, 4.14.4.264 does not allow me to blacklist individual items from JEI if they are a variant with NBT - it blacklists all of that item instead of just that NBT tag specifically. Is this possibly caused by the same reason, and is this behaviour intentional?

Reference: https://github.com/Darkhax-Minecraft/ItemStages/issues/42

#### Versions:
Previous JEI: 4.13.1.223
New JEI: 4.14.4.264
ItemStages: 2.0.46

**Comment by mezz** [comment]
Thanks for the report!
I added a pretty big hack to get around issues with enchanted books desynchronizing when connecting to servers, so they are implemented specially right now in 1.12.
I need to look into making sure they work like normal items for mods like ItemStages and CraftTweaker that want to hide them.

**Comment by mezz** [comment]
I have pushed a fix for this issue. 
I tested with SevTech and now you should see something like this in the log:
`[jei]: Ingredients are being removed at runtime: 135 net.minecraft.enchantment.EnchantmentData`
